src/configuration/ApiConfiguration.ts

```diff
@@ -0,0 +1,150 @@
+export interface CodeDetails {
+    componentCode: number;
+    stateMachineCode: number;
+}
+
+export interface PublisherDetails {
+    eventCode: number;
+    routingKey: string;
+}
+
+export enum SubscriberEventType {
+    Update,
+    Error
+}
+
+export interface ApiConfiguration {
+    getCodes(componentName: string, stateMachineName: string): CodeDetails;
+    getPublisherDetails(componentCode: number, stateMachineCode: number, messageType: string): PublisherDetails;
+    getSubscriberTopic(componentCode: number, stateMachineCode: number, type: SubscriberEventType): string;
+    getSnapshotTopic(componentCode: number): string;
+    getStateName(componentCode: number, stateMachineCode: number, stateCode: number): string;
+    containsPublisher(componentCode: number, stateMachineCode: number, messageType: string): boolean;
+    containsSubscriber(componentCode: number, stateMachineCode: number, type: SubscriberEventType): boolean;
+}
+
+export class DefaultApiConfiguration implements ApiConfiguration {
+
+    private _config: any;
+
+    constructor(configObject: any) {
+        this._config = configObject;
+        if (!this.validate()) {
+            throw new Error("invalid configuration");
+        }
+    }
+
+    private validate(): boolean {
+        return this._config
+            && this._config.deployment
+            && this._config.deployment.codesConverter
+            && this._config.deployment.clientAPICommunication;
+    }
+
+    getCodes(componentName: string, stateMachineName: string): CodeDetails {
+        const component = this.getComponentDetails(component => component.$.name === componentName, componentName);
+        const stateMachine = this.getStateMachineDetails(component, stm => stm.$.name === stateMachineName, stateMachineName);
+
+        return { componentCode: Number(component.$.id), stateMachineCode: Number(stateMachine.$.id) };
+    }
+
+    private getComponentDetails(filter: (component: any) => boolean, componentDescription?: string) {
```

Comment:
1. return type missing? I personally like adding them to all methods, private included, it makes the code generally easier to read.
2. shoun't this method be called something like "findComponent(filter)" ? of "firstComponent..."